src/ts/components/timelinemarkershandler.ts

```diff
@@ -0,0 +1,209 @@
+import { Timeout } from '../timeout';
+
+export interface MarkersConfig extends ComponentConfig {
+  /**
+   * Used for seekBar marker snapping range percentage
+   */
+  snappingRange?: number;
+}
+
+export class TimelineMarkersHandler {
+  private markersContainer: DOM;
+  private timelineMarkers: SeekBarMarker[];
+  private player: PlayerAPI;
+  private uimanager: UIInstanceManager;
+  private pausedTimeshiftUpdater: Timeout;
+  private getSeekBarWidth: () => number;
+  protected config: MarkersConfig;
+
+  constructor(config: MarkersConfig, getSeekBarWidth: () => number, markersContainer: DOM) {
+    this.config = config;
+    this.getSeekBarWidth = getSeekBarWidth;
+    this.markersContainer = markersContainer;
+  }
+
+  public configure(player: PlayerAPI, uimanager: UIInstanceManager) {
+    this.player = player;
+    this.uimanager = uimanager;
+    this.configureMarkers();
+  }
+
+  private configureMarkers(): void {
+    // Remove markers when unloaded
+    this.player.on(this.player.exports.PlayerEvent.SourceUnloaded, () => this.clearMarkers());
+    // Update markers when the size of the seekbar changes
+    this.player.on(this.player.exports.PlayerEvent.PlayerResized, () => this.updateMarkersDOM());
+
+    this.player.on(this.player.exports.PlayerEvent.SourceLoaded, () => {
+      if (this.player.isLive()) {
+        // Update marker position as timeshift range changes
+        this.player.on(this.player.exports.PlayerEvent.TimeChanged, () => this.updateMarkers());
+        // Update marker postion when paused as timeshift range changes
+        this.configureLivePausedTimeshiftUpdater(() => this.updateMarkers());
+      }
+    });
+    this.uimanager.getConfig().events.onUpdated.subscribe(() => this.updateMarkers());
+    this.uimanager.onRelease.subscribe(() => this.uimanager.getConfig().events.onUpdated.unsubscribe(() => this.updateMarkers()));
+
+    // Init markers at startup
+    this.updateMarkers();
+  }
+
+  public getMarkerAtPosition(percentage: number): SeekBarMarker | null {
+    const snappingRange = this.config.snappingRange;
+
+    if (this.timelineMarkers.length > 0) {
+      this.timelineMarkers.forEach(marker => {
+        // Handle interval markers
+        if (marker.duration > 0
+          && percentage >= marker.position - snappingRange
+          && percentage <= marker.position + marker.duration + snappingRange) {
+          return marker;
+        }
+        // Handle position markers
+        else if (percentage >= marker.position - snappingRange
+          && percentage <= marker.position + snappingRange) {
+          return marker;
+        }
+      });
+    }
+
+    return null;
+  }
+
+  private clearMarkers(): void {
+    this.timelineMarkers = [];
+    this.markersContainer.empty();
+  }
+
+  private removeMarker(marker: TimelineMarker): void {
+    this.uimanager.getConfig().metadata.markers = this.uimanager.getConfig().metadata.markers.filter(_marker => marker !== _marker);
```

Comment:
Do we really want to remove marker from the configuration it self? I think we should rather map it for internal manipulation.

Reply:
This is a single point of truth for all markers in this codebase.